### PR TITLE
Feature/overactuated pullup

### PR DESCRIPTION
**Pull request by kajabo**
Includes 6dof trajectory capabilities for fully- and overactuated systems.
Currently compatible with feature/6D_extension of mav_comm

### CODE REVIEW

File: mav_trajectory_generation/src/trajectory.cpp [hunk 1]
```diff
@@ -255,6 +259,14 @@ bool Trajectory::addTrajectories(const std::vector<Trajectory>& trajectories,
   return true;
 }
 
+bool Trajectory::offsetTrajectory(const Eigen::VectorXd& A_r_B) {
+  for (Segment& s : segments_) {
+    if (!s.offsetSegment(A_r_B)) return false;
```
Comment (michaelpantic) [inline mav_trajectory_generation/src/trajectory.cpp]: Warning, nitpick:
Could lead to weird results if not all segments are transformed. 
I know, based on the code in offsetSegment this should not happen, as A_r_B either works for all or none of the segments. But it's more convoluted to read, as it's not necessary.

How about adding the same check for dimension here and just do the check once in the beginning of the function? Then it's more clear what happens.

File: mav_trajectory_generation/src/trajectory.cpp [hunk 2]
```diff
@@ -271,6 +283,43 @@ Vertex Trajectory::getGoalVertex(int max_derivative_order) const {
...
+  CHECK_NOTNULL(yaw_vertices);
+  const std::vector<size_t> kPosDimensions = {0, 1, 2};
+  const std::vector<size_t> kYawDimensions = {3};
+  const int kMaxDerivativeOrder =
+      std::max(max_derivative_order_pos, max_derivative_order_yaw);
+  pos_vertices->resize(segments_.size() + 1, Vertex(3));
+  yaw_vertices->resize(segments_.size() + 1, Vertex(1));
+
+  // Start vertex.
+  Vertex temp_vertex(4);
```
Comment (michaelpantic) [inline mav_trajectory_generation/src/trajectory.cpp]: Will this work for D==6?

File: mav_trajectory_generation/include/mav_trajectory_generation/io.h [hunk 3]
```diff
@@ -21,11 +21,24 @@
 #ifndef MAV_TRAJECTORY_GENERATION_YAML_IO_H_
 #define MAV_TRAJECTORY_GENERATION_YAML_IO_H_
 
+#include <yaml-cpp/yaml.h>
```
[the remark this answers is not in the document]
Comment (michaelpantic) [reply mav_trajectory_generation/include/mav_trajectory_generation/io.h]: Nevermind, yaml-cpp seems to have been linked already before, so I'd suggest we leave it :-)

**Comment by rikba** [comment]
Please merge master into this branch first to view new changes.

File: mav_trajectory_generation/include/mav_trajectory_generation/vertex.h [hunk 4]
```diff
@@ -120,7 +120,8 @@ std::ostream& operator<<(std::ostream& stream,
 // required to get from one vertex to the next. Uses the current preferred
 // method.
 std::vector<double> estimateSegmentTimes(const Vertex::Vector& vertices,
-                                         double v_max, double a_max);
+                                         double v_max, double a_max,
+                                         double t_min = 0.0);
```
[the remark this answers is not in the document]
Comment (kajabo) [reply mav_trajectory_generation/include/mav_trajectory_generation/vertex.h]: Good points! Working on a more natural formulation for 6dof.

**Comment by kajabo** [comment]
@michaelpantic @rikba master now merged in, sorry!
Also: added functionality to append segments with different times, scaling polynomials to the longer segment time. In the 6dof example, velocity and acceleration limits are now indepenty set for translation and rotation.

File: mav_trajectory_generation/src/trajectory.cpp (hunk 2 above)
Comment (kajabo) [reply mav_trajectory_generation/src/trajectory.cpp]: nope! added new function overload for generic case: [20bdcd8](https://github.com/ethz-asl/mav_trajectory_generation/pull/97/commits/93cdeebbf39664a71e5368c128eb478bc21f23b9)

**Comment by rikba** [comment]
please test this

**Comment by rikba** [comment]
test this please

**Pull request merged**